### PR TITLE
Feature/core 744 add search functionality

### PR DESCRIPTION
**Pull request by karladler**

### CODE REVIEW

File: dapps/assets/src/components/SearchService.ts [hunk 1]
```diff
@@ -17,4 +17,86 @@
   the following URL: https://evan.network/license/
 */
 
-export class SearchService {}
+import axios from 'axios';
+import { utils } from '@evan.network/api-blockchain-core';
+
+let agentUrl = 'https://search.test.evan.network/api/smart-agents' // TODO: prod/dev switch
```
Comment (S3bb1) [inline dapps/assets/src/components/SearchService.ts]:
```suggestion
let agentUrl;
```

define the searchUrl when the runtime is available

File: dapps/assets/src/components/SearchService.ts [hunk 2]
```diff
@@ -17,4 +17,86 @@
...
+
+class SearchService {
+  searchUrl = `${agentUrl}/search`;
+  runtime = null;
+
+  constructor(runtime: any) {
+    if (!runtime) {
+      throw new Error('Initialized search Service without runtime.');
+    }
+    this.runtime = runtime;
```
Comment (S3bb1) [inline dapps/assets/src/components/SearchService.ts:60]:
```suggestion
    this.runtime = runtime;
    agentUrl = `https://search${runtime.environment === 'testcore' ? '.test' : ''}.evan.network/api/smart-agents' 
```

File: dapps/assets/src/routes.ts [hunk 3]
```diff
@@ -30,9 +30,23 @@ const routeRegistration: Array<RouteRegistrationInterface> = [
...
+          meta: { type: 'twins' }
+        }, {
+          path: '',
+          component: DigitalTwinsComponent,
+          meta: { type: 'twins' }
+        }]
+      },
       { name:'contacts', path: 'contacts', component: ContactsComponent },
-      { name:'search', path: 'search/:query', component: SearchResultComponent },
+      // { name:'search', path: 'search/:query', component: SearchResultComponent },
```
Comment (S3bb1) [inline dapps/assets/src/routes.ts]:
```suggestion
```

obsolete?

File: dapps/assets/src/components/SearchService.ts [hunk 4]
```diff
@@ -17,4 +17,86 @@
...
+    params.searchTerm = !params.searchTerm || params.searchTerm === '*'
+      ? '*'
+      :`*${ params.searchTerm }*`;
+
+    const { data: { result, total }} = await axios.get(`${ this.searchUrl }/${ type }`, {
+      headers: {
+        'Authorization': authHeaders,
+      },
+      params,
+    }) as unknown as SearchResponse;
```
Comment (S3bb1) [inline dapps/assets/src/components/SearchService.ts]: is this the correct definition? looks strange

File: dapps/assets/src/i18n/de.json [hunk 5]
```diff
@@ -7,6 +7,10 @@
       "my-own": "Eigene",
       "favorites": "Favoriten",
       "all": "Alle"
+    },
+    "search": {
+      "searched-for": "Sie sichten nach",
```
Comment (S3bb1) [inline dapps/assets/src/i18n/de.json]:
```suggestion
      "searched-for": "Sie suchten nach",
```

File: dapps/assets/src/i18n/de.json [hunk 6]
```diff
@@ -7,6 +7,10 @@
...
+      "no-results": "Leider keine Ergebnisse zu Ihrer Anfrage gefunden"
```
Comment (S3bb1) [inline dapps/assets/src/i18n/de.json]:
```suggestion
      "no-results": "Leider wurden keine Ergebnisse zu Ihrer Anfrage gefunden"
```

sounds better

File: dapps/assets/src/components/SearchService.ts (hunk 4 above)
Comment (karladler) [reply dapps/assets/src/components/SearchService.ts]: not sure, told me what the linter said to do 🤣  but it works and linter didn't complained any more. Axios returns always any type promise.

Comment (datrinh) [reply dapps/assets/src/components/SearchService.ts]: You can type the return from axios like so:

```
axios.get<SearchResponse>(`${ this.searchUrl }/${ type }` ...
```

**Review by datrinh — CHANGES_REQUESTED** [review]

File: dapps/assets/src/components/SearchService.ts (hunk 4 above)
Comment (karladler) [reply dapps/assets/src/components/SearchService.ts]: da bekommt er das destructuring nicht hin... strange

Edit: habs umgebaut, so geht's. Doppeltes destructuring war ihm wohl zu viel :D

**Review by Tschuck — APPROVED** [review]

**Pull request merged**